### PR TITLE
Implement setting and clearing debug app

### PR DESCRIPTION
**Pull request by bacecek**
A bit more information here:
https://developer.android.com/tools/adb#am

### CODE REVIEW

**Comment by bacecek** [comment]
Looks like the problem in tests is timeout and it should be increased
DeepSource shows problems that occur in whole codebase, so it's not only my PR problem
And the error in installation verification action is `error: The 'psutil' distribution was not found and is required by adb-enhanced`, which is also not seems to be my PR problem

**Comment by ashishb** [comment]
Thanks for the contribution.

Please ignore `DeepSource shows problems that occur in whole codebase, so it's not only my PR problem` - I disabled it already.


> And the error in installation verification action is error: The 'psutil' distribution was not found and is required by adb-enhanced, which is also not seems to be my PR problem

This is a weird issue that happens on my PRs as well!

> Looks like the problem in tests is timeout and it should be increased

I will increase the timeouts.
I think the timeouts are reflective of existing cached AVDs that have been evicted.

File: adbe/adb_enhanced.py [hunk 1]
```diff
@@ -2090,6 +2090,22 @@ def toggle_location(turn_on):
     execute_adb_shell_settings_command(cmd)
 
 
+@ensure_package_exists
+def set_debug_app(app_name, wait_for_debugger, persistent):
```
Comment (ashishb) [inline adbe/adb_enhanced.py:2094]: Do you know in which Android API version was `am set-debug-app` introduced?

I use `_error_if_min_version_less_than(<version>)` to error out if the command is going to fail anyways.

**Review by ashishb — APPROVED** [review]

File: adbe/adb_enhanced.py (hunk 1 above)
Comment (bacecek) [reply adbe/adb_enhanced.py:2094]: As I can see in sources, `set-debug-app` and `clear-debug-app` have been added in year of 2011: https://cs.android.com/android/_/android/platform/frameworks/base/+/d8c98fee02b5498270344727d1289004d98c0e73
Unfortunately, I have no idea how to find the actual Android version in specific commit: https://cs.android.com/android/_/android/platform/frameworks/base/+/d8c98fee02b5498270344727d1289004d98c0e73:. Please, let me know, if you know how to find it.
As I can see here: https://cs.android.com/android/_/android/platform/frameworks/base/+/d8c98fee02b5498270344727d1289004d98c0e73:api/current.txt;l=14409, the actual API version is 15. Unfortunately, I can't find emulator system images of API 14 or 15 on arm64 to check that.

Comment (ashishb) [reply adbe/adb_enhanced.py:2094]: No worries, these are old enough that practically any device one would test will support them.

**Comment by ashishb** [comment]
@bacecek why would this test be so slow
`692.69s call     tests/adbe_tests.py::test_debug_app` 
Reference: https://github.com/ashishb/adb-enhanced/actions/runs/8588180300/job/23532695878?pr=234

**Comment by bacecek** [comment]
> @bacecek why would this test be so slow `692.69s call tests/adbe_tests.py::test_debug_app` Reference: https://github.com/ashishb/adb-enhanced/actions/runs/8588180300/job/23532695878?pr=234

Oh, 11 minutes, this is insane. 
On my computer and API 21 emulator it works well (about 5 seconds).
I will try to remove some tests.

**Comment by bacecek** [comment]
@ashishb Are there any differences in emulator set up for AdbeInstallTests and AdbeUnitTests checks?
I see that two checks on API 21 emulators takes different times to run:
https://github.com/ashishb/adb-enhanced/actions/runs/8591828733/job/23541084021?pr=234 - it's ok
https://github.com/ashishb/adb-enhanced/actions/runs/8591828735/job/23541083983?pr=234 - it timeouts on my tests.
Do you have any ideas?

**Comment by ashishb** [comment]
`107.13s call     tests/adbe_tests.py::test_debug_app_set_fail` - https://github.com/ashishb/adb-enhanced/actions/runs/8592107146/job/23541680568?pr=234 much better but I still wonder why this call is so slow.


> I see that two checks on API 21 emulators takes different times to run:
Fixed in https://github.com/ashishb/adb-enhanced/pull/241

**Comment by bacecek** [comment]
> This is much faster than earlier but I still wonder why this call is so slow.

I'll try to have a look

**Comment by bacecek** [comment]
Unfortunately, I have no idea why tests setting `-p` and `-w`  flags and trying to set non existence app takes so much time. On my local machine they all takes less than a second to run. And as I can see, there is a different behavior on different API versions, so maybe it's a problem of the exact emulator.
I left single test that checks plain `set` and `clear`.

**Pull request merged**